Bug: ошибка компиляции из-за неопределенной работы функции слайдера

1>D:\Unreal Projects\Game\G2I-Game\Source\G2I\Private\Gameplay\G2ISliderLampComponent.cpp(34,17): error C2668: 'UMaterialInstanceDynamic::SetVectorParameterValue': ambiguous call to overloaded function
1>  DynamicMaterial->SetVectorParameterValue("EmissiveColor", {0.0f, 0.0f, 0.0f, 1.0f});
1>                 ^
1>D:\UE_5.6\Engine\Source\Runtime\Engine\Public\Materials\MaterialInstanceDynamic.h(113,14): note: could be 'void UMaterialInstanceDynamic::SetVectorParameterValue(FName,const FVector4 &)'
1>  inline void SetVectorParameterValue(FName ParameterName, const FVector4& Value) { SetVectorParameterValue(ParameterName, FLinearColor(Value)); }
1>              ^
1>D:\UE_5.6\Engine\Source\Runtime\Engine\Public\Materials\MaterialInstanceDynamic.h(109,18): note: or       'void UMaterialInstanceDynamic::SetVectorParameterValue(FName,FLinearColor)'
1>  ENGINE_API void SetVectorParameterValue(FName ParameterName, FLinearColor Value);
1>                  ^
1>D:\Unreal Projects\Game\G2I-Game\Source\G2I\Private\Gameplay\G2ISliderLampComponent.cpp(34,17): note: while trying to match the argument list '(const char [14], initializer list)'
1>  DynamicMaterial->SetVectorParameterValue("EmissiveColor", {0.0f, 0.0f, 0.0f, 1.0f});
1>                 ^
1>D:\Unreal Projects\Game\G2I-Game\Source\G2I\Private\Gameplay\G2ISliderLampComponent.cpp(79,20): error C2668: 'UMaterialInstanceDynamic::SetVectorParameterValue': ambiguous call to overloaded function
1>        DynamicMaterial->SetVectorParameterValue("EmissiveColor", {0.0f, 0.0f, 0.0f, 1.0f});
1>                       ^
1>D:\UE_5.6\Engine\Source\Runtime\Engine\Public\Materials\MaterialInstanceDynamic.h(113,14): note: could be 'void UMaterialInstanceDynamic::SetVectorParameterValue(FName,const FVector4 &)'
1>  inline void SetVectorParameterValue(FName ParameterName, const FVector4& Value) { SetVectorParameterValue(ParameterName, FLinearColor(Value)); }
1>              ^
1>D:\UE_5.6\Engine\Source\Runtime\Engine\Public\Materials\MaterialInstanceDynamic.h(109,18): note: or       'void UMaterialInstanceDynamic::SetVectorParameterValue(FName,FLinearColor)'
1>  ENGINE_API void SetVectorParameterValue(FName ParameterName, FLinearColor Value);
1>                  ^
1>D:\Unreal Projects\Game\G2I-Game\Source\G2I\Private\Gameplay\G2ISliderLampComponent.cpp(79,20): note: while trying to match the argument list '(const char [14], initializer list)'
1>        DynamicMaterial->SetVectorParameterValue("EmissiveColor", {0.0f, 0.0f, 0.0f, 1.0f});
1>                       ^
1>Trace written to file C:/Users/Пользователь/AppData/Local/UnrealBuildTool/Log.uba with size 5.0kb
1>Total time in Unreal Build Accelerator local executor: 3.01 seconds
1>
1>Result: Failed (OtherCompilationError)
1>Total execution time: 3.46 seconds
1>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.MakeFile.Targets(44,5): error MSB3073: The command "D:\UE_5.6\Engine\Build\BatchFiles\Build.bat G2IEditor Win64 Development -Project="D:\Unreal Projects\Game\G2I-Game\G2I.uproject" -WaitMutex -FromMsBuild -architecture=x64" exited with code 6.
1>Done building project "G2I.vcxproj" -- FAILED.

--- a/Source/G2I/Private/Gameplay/G2ISliderLampComponent.cpp
+++ b/Source/G2I/Private/Gameplay/G2ISliderLampComponent.cpp
@@ -31,7 +31,7 @@ void UG2ISliderLampComponent::BeginPlay()
 	}
 	
 	DynamicMaterial->SetVectorParameterValue("BaseColor", LampColor);
-	DynamicMaterial->SetVectorParameterValue("EmissiveColor", {0.0f, 0.0f, 0.0f, 1.0f});
+	DynamicMaterial->SetVectorParameterValue("EmissiveColor", FLinearColor{0.0f, 0.0f, 0.0f, 1.0f});
 	DynamicMaterial->SetScalarParameterValue("EmissiveCoeff", 0.0f);
 	
 	if (LampMesh)
@@ -76,7 +76,7 @@ void UG2ISliderLampComponent::ChangeIntensity(int IntensityChangeDir, float Targ
 			World->GetTimerManager().ClearTimer(IntensityColorTimer);
 			if (FMath::IsNearlyEqual(EmissiveCoeff, 0.0f))
 			{
-				DynamicMaterial->SetVectorParameterValue("EmissiveColor", {0.0f, 0.0f, 0.0f, 1.0f});
+				DynamicMaterial->SetVectorParameterValue("EmissiveColor", FLinearColor{0.0f, 0.0f, 0.0f, 1.0f});
 			}
 		}
 	}	

